### PR TITLE
External signer documentation for staking CLI

### PR DESCRIPTION
**Pull request by KPrasch**
Introduction of external signing documentation focused on clef and the staking CLI.

### CODE REVIEW

File: docs/source/guides/network_node/staking_guide.rst [hunk 1]
```diff
@@ -102,6 +102,109 @@ All staking-related operations done by Staker are performed through the ``nucyph
...
+
+    # Create a new stake using inline signer and provider values
+    $ nucypher stake create --signer ~/clef/clef.ipc --provider ~/.ethereum/goerli/geth.ipc
+
+
+Interacting with clef
+*********************
+
+Requests for account management, and signing will be directed at clef, with a 10 second timeout.
+Be alert for user-interactive requests from the clef CLI.
```
Comment (derekpierre) [inline docs/source/guides/network_node/staking_guide.rst]:
```suggestion
Be alert for user-interactive requests and confirmations from the clef CLI.
```

File: docs/source/guides/network_node/staking_guide.rst [hunk 2]
```diff
@@ -102,6 +102,109 @@ All staking-related operations done by Staker are performed through the ``nucyph
...
+.. important::
+
+    External signing support is an experimental feature and under active development.
+
+Motivation
+**********
+
+Instead of running a full ethereum node for both transaction signing and broadcast, an external signer can be specified
+via command line independent from the provider. This allows stakers and other network users to use hardware and software
+wallets without needing to run an ethereum node at all.
```
[the remark this answers is not in the document]
Comment (KPrasch) [reply docs/source/guides/network_node/staking_guide.rst]: Give it another look with the added heading description under "Staking".  I want to be careful not to take on the tone of a report of *what changed*.  

For example:
> Currently, a full ethereum node needs to be run to perform on-chain transactions.

Is not really true anymore.

[the remark this answers is not in the document]
Comment (derekpierre) [reply docs/source/guides/network_node/staking_guide.rst]: Sorry that was supposed to be "Previously".

[the remark this answers is not in the document]
Comment (derekpierre) [reply docs/source/guides/network_node/staking_guide.rst]: Fair point - updated suggestion.

[the remark this answers is not in the document]
Comment (KPrasch) [reply docs/source/guides/network_node/staking_guide.rst]: Have another look :-)

File: docs/source/guides/network_node/staking_guide.rst [hunk 3]
```diff
@@ -102,6 +102,109 @@ All staking-related operations done by Staker are performed through the ``nucyph
...
+Clef can use hardware wallets (ledger and trezor) over USB, or geth formatted private keys
+by specifying the keystore directory path:
+
+.. code:: bash
+
+    $ clef --keystore <PATH TO GOERLI KEYSTORE> --chainid 5 --advanced
+
+
+- <PATH TO KEYSTORE> - The path to the directory containing geth-formatted private key files.
+  the default for linux is `~/.ethereum/goerli/keystore`.
```
[the remark this answers is not in the document]
Comment (KPrasch) [reply docs/source/guides/network_node/staking_guide.rst]: :heavy_check_mark:

File: docs/source/guides/network_node/staking_guide.rst (hunk 1 above)
Comment (KPrasch) [reply docs/source/guides/network_node/staking_guide.rst]: :heavy_check_mark:

File: docs/source/guides/network_node/staking_guide.rst [hunk 4]
```diff
@@ -102,6 +102,109 @@ All staking-related operations done by Staker are performed through the ``nucyph
...
+
+Clef is typically installed alongside geth.  If you already have geth installed on you system you
+may already have clef installed.  To check for an existing installation run:
+
+.. code:: bash
+
+    $ clef --version
+    Clef version 0.0.0
+
+Next, initialize Clef with your chosen password to encrypt the master seed:
```
[the remark this answers is not in the document]
Comment (KPrasch) [reply docs/source/guides/network_node/staking_guide.rst]: :heavy_check_mark:

File: docs/source/guides/network_node/staking_guide.rst [hunk 5]
```diff
@@ -102,9 +99,19 @@ All staking-related operations done by Staker are performed through the ``nucyph
 Staking
 --------
 
+Staking transactions can be broadcast using either a local or remote ethereum node.
+By default transaction signing requests are forwarded to the configured ethereum provider;
+This is the typical configuration for locally or independently run ethereum nodes.
+In order to use a remote ethereum provider (Alchemy, Infura, Public Remote Node) an external transaction signing client
+(like clef or geth) is needed separate from the broadcasting node.
```
Comment (derekpierre) [inline docs/source/guides/network_node/staking_guide.rst:106]:
```suggestion
(e.g. clef or geth) is needed separate from the broadcasting node.
```

File: docs/source/guides/network_node/staking_guide.rst [hunk 6]
```diff
@@ -102,9 +99,19 @@ All staking-related operations done by Staker are performed through the ``nucyph
 Staking
 --------
 
+Staking transactions can be broadcast using either a local or remote ethereum node.
+By default transaction signing requests are forwarded to the configured ethereum provider;
+This is the typical configuration for locally or independently run ethereum nodes.
+In order to use a remote ethereum provider (Alchemy, Infura, Public Remote Node) an external transaction signing client
```
Comment (derekpierre) [inline docs/source/guides/network_node/staking_guide.rst:105]:
```suggestion
In order to use a remote ethereum provider (e.g. Alchemy, Infura, Public Remote Node) an external transaction signing client
```

**Pull request merged**